src/main/java/me/zeroeightsix/kami/util/VectorUtil.java

```diff
@@ -0,0 +1,75 @@
+package me.zeroeightsix.kami.util;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class VectorUtil {
+
+    public static double getDistance(Vec3d vecA, Vec3d vecB) {
+        return MathHelper.sqrt((vecA.x - vecB.x) * (vecA.x - vecB.x) + (vecA.y - vecB.y) * (vecA.y - vecB.y) + (vecA.z - vecB.z) * (vecA.z - vecB.z));
+    }
+
+    public static List<Vec3d> extendVec(Vec3d startVec, Vec3d destinationVec, int steps) {
+        List<Vec3d> returnList = new ArrayList<>();
+        double stepDistance = getDistance(startVec, destinationVec) / steps;
+
+        for (int i = 0; i < Math.max(steps, 1) + 1; i++) {
+            returnList.add(advanceVec(startVec, destinationVec, stepDistance * i));
+        }
+
+        return returnList;
+    }
+
+    public static Vec3d advanceVec(Vec3d startVec, Vec3d destinationVec, double distance) {
```

Comment:
Advancing a vec? What? I still have 0 explanation of the purpose or reason behind any of these functions. Once again, ADD A JAVA DOC!!!!